File: core/src/main/java/net/remmintan/mods/minefortress/core/interfaces/server/IBlueprintEditingWorld.java
package net.remmintan.mods.minefortress.core.interfaces.server;

import net.remmintan.mods.minefortress.core.interfaces.blueprints.BlueprintGroup;

public interface IBlueprintEditingWorld {
    String getFileName();

    void setFileName(String fileName);

    int getFloorLevel();

    void setFloorLevel(int floorLevel);

    BlueprintGroup getBlueprintGroup();

    void setBlueprintGroup(BlueprintGroup blueprintGroup);

    void enableSaveStructureMode();

    void disableSaveStructureMode();
}

File: src/core/java/net/remmintan/mods/minefortress/core/interfaces/server/IBlueprintEditingWorld.kt
package net.remmintan.mods.minefortress.core.interfaces.server

import net.remmintan.mods.minefortress.core.interfaces.blueprints.BlueprintGroup

interface IBlueprintEditingWorld {
    var blueprintId: String?
    var blueprintName: String?
    var floorLevel: Int
    var blueprintGroup: BlueprintGroup?

    fun enableSaveStructureMode()

    fun disableSaveStructureMode()
}
